jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ServletLifeCycleTest.java

```diff
@@ -60,8 +60,8 @@ public void testLifeCycle() throws Exception
         context.getObjectFactory().addDecorator(new TestDecorator());
 
         ServletHandler sh = context.getServletHandler();
-        sh.addListener(new ListenerHolder(TestListener.class));
-        context.addEventListener(context.getServletContext().createListener(TestListener2.class));
+        sh.addListener(new ListenerHolder(TestListener.class)); //added directly to ServletHandler
+        context.addEventListener(context.getServletContext().createListener(TestListener2.class));//create,decorate and add listener to context - no holder!
```

Comment:
What is this comment all about?
Does using ServletHolder via ServletContextHandler change the behavior in some way?

Reply:
The decoration of the listener happens when the listener instance is created. If you call ServletContext.createListener(Class), then the instance is created at that point and the decoration happens at that point.  If you call ServletHandler.addListener(ListenerHolder), the ListenerHolder delays the creation of the instance until the holder is started, thus the decoration and adding of the listener instance to the ServletHandler happens at a different point.

Reply:
ah.